agent/agent.go

```diff
@@ -160,6 +160,17 @@ func (p *SnapProcessor) isDataNull(data interface{}) bool {
 	return false
 }
 
+func (p *SnapProcessor) getCacheKey(mt plugin.Metric) string {
+	namespaces := mt.Namespace.Strings()
+	cacheKey := strings.Join(namespaces, "/")
+	isContainerNamespaces := strings.HasPrefix(cacheKey, "intel/docker/")
+	if nodename, ok := mt.Tags["nodename"]; ok && !isContainerNamespaces {
```

Comment:
What happens if it's a container metric? Looks like we fail back on just using the metric name, but we need to use container name instead.